x/blob/types/blob_tx.go

```diff
@@ -109,6 +111,27 @@ func BlobTxSharesUsed(btx tmproto.BlobTx) int {
 	return sharesUsed
 }
 
+func BlobFromProto(p *tmproto.Blob) core.Blob {
+	if p == nil {
+		return core.Blob{}
+	}
+
+	if p.ShareVersion > math.MaxUint8 {
+		return core.Blob{}
+	}
+
+	if p.NamespaceVersion > math.MaxUint8 {
+		return core.Blob{}
+	}
```

Comment:
[non blocking question]
do we know the thinking around adding these? was it just copied over, or is it that we're more likely to notice empty blobs than an overflow?

Reply:
Around adding what: the function `BlobFromProto` or the checks for overflow? 

1. `BlobFromProto` was copied over b/c a recent change in celestia-app started using this exported type in celestia-core but a recent change in celestia-core removed it.
2. The checks for overflow are to catch overflows because the fields in the proto type can be larger than uint8

Reply:
I see the overflow, but I'm curious if we should instead throw an error, since returning an empty blob for an otherwise valid blob might cause a bug. I don't think we'd ever have issues if we only use this function in the way we are now, but I could see it becoming an issue is usage, by us or otherwise, changes.

Reply:
+100 returning an error seems safer

Reply:
Resolved by https://github.com/celestiaorg/celestia-app/pull/1664/commits/14d4c7399debcb6f8164bfb0ae1e7ba41bc2b24a